### PR TITLE
Update SDK and start using tools/firmware

### PR DESCRIPTION
**Pull request by kasperl**

### CODE REVIEW

File: cmd/jag/commands/firmware.go [hunk 1]
```diff
@@ -150,48 +139,51 @@ func BuildFirmwareImage(ctx context.Context, id string, name string, wifiSSID st
...
-		binTmpFile.Close()
 		return nil, err
 	}
 
-	_, err = io.Copy(binTmpFile, binFile)
-	binTmpFile.Close()
-	binFile.Close()
-	if err != nil {
+	// TODO(kasper): Can we generate this in a nicer way?
+	wifiConfig := "{ \"wifi.ssid\": \"" + wifiSSID + "\", \"wifi.password\": \"" + wifiPassword + "\" }"
```
Comment (floitsch) [inline cmd/jag/commands/firmware.go:160]: Maybe?
```suggestion
	wifiConfig := fmt.Sprintf("{ \"wifi.ssid\": \"%s\", \"wifi.password\": \"%s"\" }", wifiSSID, wifiPassword)
```
Alternatively, obviously, using JSON.

File: Makefile [hunk 2]
```diff
@@ -105,17 +105,20 @@ $(BUILD_DIR)/image/partitions.csv: $(TOIT_PATH)/toolchains/esp32/partitions.csv
...
 image: $(BUILD_DIR)/image/toit.elf
-image: $(BUILD_DIR)/image/toit.bin
 image: $(BUILD_DIR)/image/bootloader/bootloader.bin
 image: $(BUILD_DIR)/image/partitions.bin
 image: $(BUILD_DIR)/image/partitions.csv
 image: $(BUILD_DIR)/image/system.snapshot
-image: $(BUILD_DIR)/image/jaguar.snapshot
 
 .PHONY: install-esp-idf
 install-esp-idf:
```
Comment (floitsch) [inline Makefile:124]: I think you want to remove `--auto-sync` from the `install-dependencies`. Otherwise the package manager won't find all packages.

**Review by floitsch — APPROVED** [review]
LGTM.

File: Makefile (hunk 2 above)
Comment (kasperl) [reply Makefile:124]: Done.

File: cmd/jag/commands/firmware.go (hunk 1 above)
Comment (kasperl) [reply cmd/jag/commands/firmware.go:160]: I'll probably change it to map -> json.stringify soon.

**Pull request merged**